File: test20_out/atmel.atmega328/PD0/PD0.js

/*
 * ======== MODULE atmel.atmega328.PD0 ========
 */

var $$c = function() {
    const PD0 = {};
    PD0.PD0 = PD0
    PD0.$name = 'atmel.atmega328.PD0';
    PD0.pollen$used = 0;
    PD0.set = new $$Ref('atmel_atmega328_PD0_set__E');
    PD0.clear = new $$Ref('atmel_atmega328_PD0_clear__E');
    PD0.toggle = new $$Ref('atmel_atmega328_PD0_toggle__E');
    PD0.get = new $$Ref('atmel_atmega328_PD0_get__E');
    PD0.makeInput = new $$Ref('atmel_atmega328_PD0_makeInput__E');
    PD0.isInput = new $$Ref('atmel_atmega328_PD0_isInput__E');
    PD0.makeOutput = new $$Ref('atmel_atmega328_PD0_makeOutput__E');
    PD0.isOutput = new $$Ref('atmel_atmega328_PD0_isOutput__E');
    PD0.targetInit = new $$Ref('atmel_atmega328_PD0_targetInit__I');
    PD0.$$hostInit = function() {
        var $$text = '';
        return $$text;
    }
    PD0.pollen__uses__ = function() {
    }
    PD0.$$privateInit = function() {
    }
    return PD0;
}
var $$u = $$c();
$units['atmel.atmega328.PD0'] = $$u;
$units.push($$u);

File: test22_out/atmel.atmega328/PD0/PD0.c

/*
 * ======== string literals ========
 */

static const char atmel_atmega328_PD0__s__fileName[] = "PD0.p";
static const char atmel_atmega328_PD0__s__0[] = "PD0";

/*
 * ======== code injections ========
 */

 #include <avr/io.h> 

/*
 * ======== function definitions ========
 */

void atmel_atmega328_PD0_set__E() {
    PORTD |= (1 << 0);
}

void atmel_atmega328_PD0_makeOutput__E() {
    DDRD &= ~(1 << 0);
}

bool atmel_atmega328_PD0_isInput__E() {
    return(  (DDRD & (1 << 0)) ? false : true  );
}

bool atmel_atmega328_PD0_isOutput__E() {
    return(  (DDRD & (1 << 0)) ? true : false  );
}

bool atmel_atmega328_PD0_get__E() {
    return(  (PORTD & (1 << 0)) ? true : false  );
}

void atmel_atmega328_PD0_clear__E() {
    PORTD &= (1 << 0);
}

void atmel_atmega328_PD0_toggle__E() {
    PORTD ^= (1 << 0);
}

void atmel_atmega328_PD0_makeInput__E() {
    DDRD &= ~(1 << 0);
}

void atmel_atmega328_PD0_targetInit__I() {
}

